BootloaderCorePkg/Tools/GenCapsuleFirmware.py

```diff
@@ -482,7 +484,7 @@ def ValidateRegistryFormatGuid(Argument):
 
     parser.add_argument('-p',  '--payload', nargs=2, action='append', type=str, required=True, help='Specify payload information including GUID, FileName')
     parser.add_argument('-k',  '--priv_key', dest='PrivKey', type=str, required=True, help='Key Id or Private RSA 2048/RSA3072 key in PEM format to sign image')
-    parser.add_argument('-a',  '--alg_hash', dest='HashType', type=str, choices=['SHA2_256', 'SHA2_384'], default='SHA2_256', help='Hash type for signing')
+    parser.add_argument('-a',  '--alg_hash', dest='HashType', type=str, choices=['SHA2_256', 'SHA2_384', None], default=None, help='Hash type for signing')

```

Comment:
Can we add a small description for the meaning of 'None'?
i.e. When None is chosen, the hashing mechanism will depend upon the key length chosen.